puma/var_vs_var.py

```diff
@@ -0,0 +1,409 @@
+        y_var_std: np.ndarray,
+        x_var_widths: np.ndarray = None,
+        key: str = None,
+        fill: bool = False,
+        **kwargs,
+    ) -> None:
+        """Initialise properties of roc curve object.
+
+        Parameters
+        ----------
+        x_var_mean : np.ndarray
+            Values for x-axis variable
+        y_var_mean : np.ndarray
+            Mean value for y-axis variable
+        y_var_std : np.ndarray
+            Std value for y-axis variable
+        x_var_widths : np.ndarray, optional
+            Std value for x-axis variable
+        key : str, optional
+            Identifier for the curve e.g. tagger, by default None
+        fill : bool, optional
+            Defines do we need to fill box around point, by default False
+        **kwargs : kwargs
+            Keyword arguments passed to `PlotLineObject`
+
+        Raises
+        ------
+        ValueError
+            If provided options are not compatible with each other
+        """
+
+        super().__init__(**kwargs)
+        if len(x_var_mean) != len(y_var_mean):
+            raise ValueError(
+                f"Length of `x_var_mean` ({len(x_var_mean)}) and `y_var_mean` "
+                f"({len(y_var_mean)}) have to be identical."
+            )
+        if len(x_var_mean) != len(y_var_std):
+            raise ValueError(
+                f"Length of `x_var_mean` ({len(x_var_mean)}) and `y_var_std` "
+                f"({len(y_var_std)}) have to be identical."
+            )
+
+        self.x_var_mean = np.array(x_var_mean)
+        self.x_var_widths = None if x_var_widths is None else np.array(x_var_widths)
+        self.y_var_mean = np.array(y_var_mean)
+        self.y_var_std = np.array(y_var_std)
+        self.key = key
+        self.fill = fill
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return (
+                np.all(self.x_var_mean == other.x_var_mean)
+                and np.all(self.y_var_mean == other.y_var_mean)
+                and np.all(self.y_var_std == other.y_var_std)
+                and self.key == other.key
+            )
+        return False
+
+    def divide(self, other, inverse: bool = False):
+        """Calculate ratio between two class objects.
+
+        Parameters
+        ----------
+        other : var_vs_var class
+            Second var_vs_var object to calculate ratio with
+        inverse : bool
+            If False the ratio is calculated `this / other`,
+            if True the inverse is calculated
+
+        Returns
+        -------
+        np.ndarray
+            Ratio
+        np.ndarray
+            Ratio error
+
+        Raises
+        ------
+        ValueError
+            If binning is not identical between 2 objects
+        """
+        if not np.array_equal(self.x_var_mean, other.x_var_mean):
+            raise ValueError("The x variables of the two given objects do not match.")
+        # TODO: python 3.10 switch to cases syntax
```

Comment:
I doubt we need this

Reply:
How we can divide and plot after, if the x-axis doesn't match?

Reply:
Sorry I just meant to remove the comment :)

Reply:
Fixed